### PR TITLE
[MMI] Add transaction-failed functional component

### PR DESCRIPTION
**Pull request by albertolive**
## Explanation

Add transaction-failed functional component to MM codebase

https://consensyssoftware.atlassian.net/browse/MMI-2955

Screenshot 2023-04-17 at 17 38 17

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue

### CODE REVIEW

**Review by DDDDDanica — DISMISSED** [review]

File: ui/components/institutional/transaction-failed-modal/transaction-failed.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        alignItems={AlignItems.center}
+        paddingLeft={4}
+        paddingRight={4}
+        style={{ flex: 1, overflowY: 'auto' }}
+      >
+        <img
+          className="transaction-failed__image"
+          src="images/icons/warning.svg"
+          alt=""
+        />
```
Comment (digiwand) [inline ui/components/institutional/transaction-failed-modal/transaction-failed.js:46]: I think we can replace this with the Warning icon component 

```
import { Icon, IconName } from '../../component-library';

.... 

<Icon name={IconName.Warning} size={IconSize.Xl} />
```

ref:
- https://metamask.github.io/metamask-storybook/?path=/story/components-componentlibrary-icon--default-story

File: ui/components/institutional/transaction-failed-modal/transaction-failed.test.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      />,
+      store,
+    );
+    const titleElement = screen.getByText(title);
+    expect(titleElement).toBeInTheDocument();
+  });
+
+  it('renders the correct title when transaction fails', () => {
+    const operationFailed = false;
+    const errorMessage = 'Something went wrong';
```
Comment (digiwand) [inline ui/components/institutional/transaction-failed-modal/transaction-failed.test.js:42]: [nit] we could provide this above the `describe` function and reuse throughout the file. we could also prepend the name with `mock` which can help indicate the value is an arbitrary value, `mockErrorMessage`

Comment (albertolive) [reply ui/components/institutional/transaction-failed-modal/transaction-failed.test.js]: Fixed. Thank you!

File: ui/components/institutional/transaction-failed-modal/index.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.transaction-failed {
+  &__description {
+    border: 1px solid lightgray;
```
[the remark this answers is not in the document]
Comment (albertolive) [reply ui/components/institutional/transaction-failed-modal/index.scss]: Fixed, Thank you!

File: ui/components/institutional/transaction-failed-modal/transaction-failed.js (hunk 1 above)
Comment (albertolive) [reply ui/components/institutional/transaction-failed-modal/transaction-failed.js]: Fixed, thank you

File: ui/components/institutional/transaction-failed-modal/index.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    border: 1px solid var(--color-border-muted);
+    max-width: 100%;
+    overflow-wrap: anywhere;
+  }
+
+  &__image {
+    width: 32px;
+    height: 32px;
+    border-radius: 50%;
+  }
```
[the remark this answers is not in the document]
Comment (albertolive) [reply ui/components/institutional/transaction-failed-modal/index.scss:12]: You are absolutely right

[the remark this answers is not in the document]
Comment (digiwand) [reply ui/components/institutional/transaction-failed-modal/index.scss]: 🥳

**Review by digiwand — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**